### PR TITLE
implement social priority [213]

### LINKED ISSUE #213: Support social priority
Config is bespoke:
- students with at least 1 friend
- students are with 0 enemies
- students are with as many friends as possible

### PR DESCRIPTION
**Pull request by Opeyem1a**
closes #213

### CODE REVIEW

**Comment by Opeyem1a** [comment]
Current dependencies on/for this PR:
* main
  * **PR #253** <a href="https://app.graphite.dev/github/pr/Teamable-Analytics/algorithms/253?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Teamable-Analytics/algorithms/253?utm_source=stack-comment).

File: tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    - fans (they like everyone in their group, but no reciprocity)
+    - haters (they hate everyone in their group but no reciprocity)
+    - outcasts (they hate everyone in their group and everyone hates them back)
+
+    * because neutral people will still also dislike outcasts because of the naive way it's programmed
+    """
+    students = []
+
+    if num_friends:
+        start_id = len(students) + 1
```
Comment (justino599) [inline tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py:44]: ids should be 0 indexed

**Review by justino599 — CHANGES_REQUESTED** [review]

File: tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py (hunk 1 above)
Comment (Opeyem1a) [reply tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py:44]: Oh wait why?

Comment (Opeyem1a) [reply tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py:44]: Usually, when I make students for tests I make their ids start at 1

Comment (justino599) [reply tests/test_api/test_ai/test_priority_algorithm/test_priority/_data.py:44]: Oh, I always use 0. I guess it doesn't really matter, just thought I'd bring it up

**Review by justino599 — APPROVED** [review]

**Pull request merged**